front/src/features/books/booksSlice.js

```diff
@@ -0,0 +1,79 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+import api from "api";
+
+export const booksSlice = createSlice({
+  name: "books",
+  initialState: {
+    data: [],
+    basket: [],
+    basketValue: 0,
+    dataStatus: "iddle",
+  },
+  reducers: {
+    fetchBooksDataStarted: (state) => {
+      state.dataStatus = "inProgress";
+    },
+    fetchBooksDataDataSucceeded: (state) => {
+      state.dataStatus = "succeeded";
+    },
+    fetchBooksDataDataFailed: (state) => {
+      state.dataStatus = "failed";
+    },
+    setData: (state, action) => {
+      state.data = action.payload.data;
+    },
+    addItemToBasket: (state, action) => {
+      const IsInBasket = state.basket.find(
+        (item) => item.id === action.payload.id
+      );
+
+      if (IsInBasket) {
+        state.basket = [
+          ...state.basket.map((element) => {
+            if (element.id === action.payload.id) {
+              return {
+                ...element,
+                value: element.value + 1,
+              };
+            }
+            return element;
+          }),
+        ];
+      } else {
+        const item = { ...action.payload, value: 1 };
+        state.basket = [...state.basket, item];
```

Comment:
state.basket.push({
id,
value: 1
})

troche krocej i prosciej

Reply:
a nie jest to zakazane modyfikowanie stanu Reduxa ? 
During...

Reply:
Jest teoretycznie poza akcjami, akcje są właśnie od tego żeby go modyfikować. On sam ogarnia historię stanu, nie trzeba tego zabezpieczać w żaden sposób

Reply:
A dobra :)

Reply:
Done.